client/src/main/kotlin/io/spine/examples/pingh/client/PinghApplication.kt

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2024, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.examples.pingh.client
+
+import io.spine.client.ConnectionConstants.DEFAULT_CLIENT_SERVICE_PORT
+import io.spine.core.UserId
+import java.util.UUID
+import kotlinx.coroutines.flow.MutableStateFlow
+
+/**
+ * Manages the logic for the Pingh app.
+ *
+ * Stores application states and allows to create different process flows.
+ *
+ * By default, application opens channel for the client
+ * to 'localhost:[50051][DEFAULT_CLIENT_SERVICE_PORT]'.
+ *
+ * @param address the address of the Pingh server.
+ * @param port the port on which the Pingh server is running.
+ */
+public class PinghApplication(
+    address: String = "localhost",
+    port: Int = DEFAULT_CLIENT_SERVICE_PORT
+) {
+    /**
+     * Enables interaction with the Pingh server.
+     */
+    internal val client = DesktopClient(address, port, generateUserId())
```

Comment:
Why do we generate a random ID here? Instead, we should go as guest until the user is logged in.

Once user is logged in, we should be using their user ID, which in turn must be the same for their GitHub username.